### PR TITLE
fix: re-implement useAuth to be used with OAuth

### LINKED ISSUE #266: Re-Implement `useAuth` to be used with OAuth

### PR DESCRIPTION
**Pull request by MaxSchaefer**
Closes #266

### CODE REVIEW

File: tasks/hooks/useAuth.tsx
Comment (use-to) [inline tasks/hooks/useAuth.tsx:1]: There is no tsx content in here. Why are you using `.tsx` over `.ts`

File: tasks/hooks/useAuth.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+const isJwtExpired = (token: string) => {
+  if (config.fakeTokenEnable) {
+    return !!tokenToUser(token)
+  }
+  const payloadBase64 = token.split('.')[1]
+  const decodedPayload = Buffer.from(payloadBase64, 'base64').toString()
+  const parsedPayload = JSON.parse(decodedPayload)
+  const exp = parsedPayload.exp
+  return true
+}
```
Comment (use-to) [inline tasks/hooks/useAuth.tsx:52]: Can you check this function again, please?

File: tasks/components/UserMenu.tsx [hunk 2]
```diff
@@ -34,14 +33,9 @@ export const UserMenu = ({
...
-  const auth = useAuth(() => router.push({ pathname: '/login', query: { back: true } }))
-
-  const user = auth.user
-  if (!user) return <></>
-
-  const logout = () => auth.logout(() => router.replace('/'))
+  if (!user) return null
 
   // TODO update this when introducing proper auth
   const settingsURL = 'https://auth.helpwave.de/ui/settings'
```
Comment (use-to) [inline tasks/components/UserMenu.tsx:41]: Thats part of this issue I guess. Is that static or should we move it to config?

**Review by use-to — CHANGES_REQUESTED** [review]

**Comment by use-to** [comment]
Looks really awesome! 🚀 
Let's do this!

File: tasks/components/UserMenu.tsx (hunk 2 above)
Comment (MaxSchaefer) [reply tasks/components/UserMenu.tsx]: fbe88e8a2f1f5a2522563f3cea3da817b9d7cb38

**Review by use-to — APPROVED** [review]

**Pull request merged**